### ISSUE TITLE
[unified-signatures] False positive on signature with optional parameter

### ISSUE THREAD

**Issue by lostfictions** [typescript-eslint/typescript-eslint#1704]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unified-signatures": "warn"
  }
}
```

```TS
export function randomInt(max: number): number;
export function randomInt(min: number, max: number): number;
export function randomInt(min: number, max?: number): number {
  // ...
}
```

**Expected Result**

No warning.

**Actual Result**

A warning is raised on the second overload:

```TS
export function randomInt(max: number): number;
export function randomInt(min: number, max: number): number;
//                                     ^^^^^^^^^^^
// These overloads can be combined into one signature with an optional parameter.
export function randomInt(min: number, max?: number): number
```

I at first thought ESLint might've been correct here, but removing that signature produces a TypeScript error at call sites:

```TS
const x = randomInt(3, 8);
//                     ^
// Expected 1 arguments, but got 2.ts(2554)
```

**Additional Info**

Seems possibly related to (but a more serious error than) #961.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.1.0` |
| `npm`                              | `6.13.0` |

**Comment by bradzacher** [comment]
You have misunderstood the guidance here.

The rule is not suggesting you remove the signature, it's suggesting you merge the signature into a single one.

When you create an overloaded declaration, TS doesn't consider the actual function declaration as one of the overloads, so by deleting the signature with 2 arguments, you've created a function which accepts 1 argument.

If you instead combine the signatures, it will work as expected.

`randomInt(min: number, max?: number)`
